cogito/check.go

```diff
@@ -0,0 +1,42 @@
+package cogito
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+// Check implements the "check" step (the "check" executable).
+//
+// From https://concourse-ci.org/implementing-resource-types.html#resource-check:
+//
+// A resource type's check script is invoked to detect new versions of the resource.
+// It is given the configured source and current version on stdin, and must print the
+// array of new versions, in chronological order (oldest first), to stdout, including
+// the requested version if it is still valid.
+//
+func Check(log hclog.Logger, in io.Reader, out io.Writer, args []string) error {
+	ci, err := NewCheckInput(in)
+	if err != nil {
+		return fmt.Errorf("check: %s", err)
+	}
+
+	log = log.Named("check")
+	log.SetLevel(hclog.LevelFromString(ci.Source.LogLevel))
+	log.Debug("started",
+		"source", ci.Source, "version", ci.Version, "environment", ci.Env, "args", args)
+	defer log.Debug("finished")
+
+	// Since there is no meaningful real version for this resource, we return always the
+	// same dummy version (we need to return a non-empty list to keep Concourse happy).
```

Comment:
This "making Concourse happy" comment is wrong.

From https://concourse-ci.org/implementing-resource-types.html#resource-check:
> The list may be empty, if there are no versions available at the source.

When the s3 resource doesn't find anything Concourse is still happy:
```sh
$ fly -t developers resource-versions -r survey-SURVEY-3502-rename-sources/artifacts-windows.s3 --json -c 1
[]
```

Reply:
I saw that part of the documentation and wondered the same.
I _think_ that when I wrote that comment, for the first version of Cogito, Concourse did want a non-empty list. Or maybe I just misunderstood something, I do not remember.

I will revisit this, thanks.

Reply:
Will be investigated in PCI-2617.